packages/core/src/components/axes/toolbar.ts

```diff
@@ -0,0 +1,495 @@
+					menuItem.id + this.overflowMenuItemId
+				);
+				if (element !== null) {
+					element.addEventListener("click", () => {
+						// call the specified function
+						menuItem.clickFunction();
+
+						// hide overflow menu
+						self.showOverflowMenu(false);
+					});
+					element.addEventListener("keyup", (e) => {
+						if (e.key === "Enter") {
+							// call the specified function
+							menuItem.clickFunction();
+						} else if (e.key === "ArrowUp") {
+							// focus on previous menu item
+							self.focusOnPreviousEnabledMenuItem(index);
+						} else if (e.key === "ArrowDown") {
+							// focus on next menu item
+							self.focusOnNextEnabledMenuItem(index);
+						}
+						// Not hide overflow menu by keyboard arrow up/down event
+					});
+				}
+			});
+
+			// default to focus on the first enabled menu item
+			self.focusOnNextEnabledMenuItem(-1);
+		}
+		event.stopImmediatePropagation();
+	}
+
+	getOverflowMenuHTML() {
+		const overflowMenuItems = this.getOverflowMenuItems();
+		// don't render whole overflow menu if no overflow menu item
+		if (!overflowMenuItems || overflowMenuItems.length === 0) {
+			return "";
+		}
+
+		let overflowMenuHtml;
+		overflowMenuHtml = `<div data-floating-menu-container="true" data-floating-menu-direction="bottom" role="main">
+			<ul class="bx--overflow-menu-options bx--overflow-menu--flip bx--overflow-menu-options--open"
+				tabindex="-1" role="${Roles.MENU}" aria-label="Menu" data-floating-menu-direction="bottom"
+				style="left:${this.overflowMenuX}px; top:${this.overflowMenuY}px;">`;
+
+		// generate html for each overflow menu items
+		overflowMenuItems.forEach((menuItem, index) => {
+			const menuItemClasses = "bx--overflow-menu-options__option".concat(
+				menuItem.shouldBeDisabled()
+					? " bx--overflow-menu-options__option--disabled" // class for disabled menu item
+					: ""
+			);
+			overflowMenuHtml += `<li class="${menuItemClasses}" role="${
+				Roles.MENU_ITEM
+			}">
+				<button class="bx--overflow-menu-options__btn"
+					data-floating-menu-primary-focus="${index === 0}"
+					tabindex="-1" index="${index}" title="${menuItem.text}"
+					id="${menuItem.id + this.overflowMenuItemId}">
+					<div class="bx--overflow-menu-options__option-content">
+						${menuItem.text}
+					</div>
+				</button>
+			</li>`;
+		});
+
+		overflowMenuHtml += `</ul></div>`;
+		return overflowMenuHtml;
+	}
+
+	getControlConfigs() {
+		const isZoomBarEnabled =
+			this.services.zoom.isZoomBarEnabled() &&
+			!this.services.zoom.isEmptyState();
+
+		const numberOfIcons = Tools.getProperty(
+			this.model.getOptions(),
+			"toolbar",
+			"numberOfIcons"
+		);
+		const controls = Tools.getProperty(
+			this.model.getOptions(),
+			"toolbar",
+			"controls"
+		);
+		const controlList = [];
+		controls.forEach((control) => {
+			let controlConfig;
+			switch (control.type) {
+				case ToolbarControlTypes.ZOOM_IN:
+					if (isZoomBarEnabled) {
+						controlConfig = this.getZoomInConfig();
```

Comment:
@theiliad 
I don't get your point.
Even you have a `getControlConfigs(control: ToolbarControlTypes)` method, you still need a `switch` in the method to return respective config based on `ToolbarControlTypes`.
Could you provide a pseudo code or example ?

Reply:
Hi, yes what I'm trying to avoid is a `getZoomInConfig()` call for a small object return. This'll result in too many methods in the future.

If we just have a `getControlConfigs()` that includes all the configs and returns the 1 we care about, then it's simpler

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/791/commits/bcee77bd714f8fab66157934d5178c6660e59e06.
Please let me know if this is not what you expected, thanks.